drivers/spi/spi_ll_stm32.c

```diff
@@ -348,15 +420,17 @@ static void spi_stm32_shift_s(SPI_TypeDef *spi, struct spi_stm32_data *data)
 /*
  * Without a FIFO, we can only shift out one frame's worth of SPI
  * data, and read the response back.
- *
- * TODO: support 16-bit data frames.
  */
 static int spi_stm32_shift_frames(SPI_TypeDef *spi, struct spi_stm32_data *data)
 {
 	uint16_t operation = data->ctx.config->operation;
 
 	if (SPI_OP_MODE_GET(operation) == SPI_OP_MODE_MASTER) {
-		spi_stm32_shift_m(spi, data);
+		if (operation & SPI_HALF_DUPLEX) {
+			spi_stm32_shift_m_half(spi, data);
```

Comment:
I wrote it the way I did to avoid making `spi_stm32_shift_m` have a bunch of cases that do different things, but if that style is better then I'm not opposed to changing it.

Reply:
Reusing spi_stm32_shift_m would indeed be more memory efficient.
Can you take this request into account ?